api/liboni/oni.c

```diff
@@ -171,48 +130,27 @@ oni_ctx oni_create_ctx()
     return ctx;
 }
 
-int oni_init_ctx(oni_ctx ctx)
+int oni_init_ctx(oni_ctx ctx, int device_index)
 {
     assert(ctx != NULL && "Context is NULL.");
     assert(ctx->run_state == UNINITIALIZED && "Context is in invalid state.");
 
     if (ctx->run_state != UNINITIALIZED)
         return ONI_EINVALSTATE;
 
-    // Open the device files
-    ctx->config.fid = open(ctx->config.path, O_RDWR | _O_BINARY);
-    if (ctx->config.fid == -1) {
-        //fprintf(stderr, "%s: %s\n", strerror(errno), ctx->config.path);
-        return ONI_EPATHINVALID;
-    }
-
-    ctx->signal.fid = open(ctx->signal.path, O_RDONLY | _O_BINARY);
-    if (ctx->signal.fid == -1) {
-        //fprintf(stderr, "%s: %s\n", strerror(errno), ctx->signal.path);
-        return ONI_EPATHINVALID;
-    }
-
-    ctx->read.fid = open(ctx->read.path, O_RDONLY | _O_BINARY);
-    if (ctx->read.fid == -1) {
-        //fprintf(stderr, "%s: %s\n", strerror(errno), ctx->read.path);
-        return ONI_EPATHINVALID;
-    }
-
-    ctx->write.fid = open(ctx->write.path, O_WRONLY | _O_BINARY);
-    if (ctx->write.fid == -1) {
-        //fprintf(stderr, "%s: %s\n", strerror(errno), ctx->write.path);
-        return ONI_EPATHINVALID;
-    }
+	int error = ctx->driver.init(ctx->driver.ctx, device_index);
```

Comment:
replace `int error` with `int rc` and remove declaration of rc below. Repeat this pattern wherever else it occurs.